rest/main.go

```diff
@@ -120,6 +120,7 @@ func serverMainPersistentConfig(fs *flag.FlagSet, flagStartupConfig *StartupConf
 				// the config is actually a 3.x config but with a genuine unknown field, therefore we should  return the
 				// original error from LoadStartupConfigFromPath.
 				if pkgerrors.Cause(upgradeError) == base.ErrUnknownField {
+					base.Errorf("automatic upgrade attempt failed, %s not recognized as legacy config format: %v", base.MD(configPath[0]), upgradeError)
```

Comment:
What do you think about this change? I'm particularly worried about users seeing this upgrade error appear for all 3.x unknown field errors, even if the config file is not a legacy config file.

```suggestion
					base.Warnf("Automatic upgrade attempt failed, %s not recognized as legacy config format: %v - continuing as startup config", base.MD(configPath[0]), upgradeError)
```

Reply:
As discussed yesterday, maybe we should rephrase the 3.x error to suggest it can be caused by one of two things.

Reply:
Based on discussion changed now to include both errors. Changed PR description to reflect example